### PR TITLE
feat: rescue-scope, bound, and correct adoption attribution counting (ADS-1023/1024/1025)

### PR DESCRIPTION
**Pull request by ParagonJenko**
## Summary

- ADS-1023: `CountAdoptedAdopters` accepts an optional `rescue_id` so an event's adoption attribution only counts adoptions at the event's own rescue.
- ADS-1024: `CountAdoptedAdopters` validates `adopter_ids` size and the date-range inputs before running any query, and `service.rescue` refuses to attribute an event with an unbounded registrant list instead of silently truncating it.
- ADS-1025: attribution now filters on a new `applications.actioned_at` column (stamped when an application reaches `approved`/`adopted`) instead of `created_at`, matching the rule the RPC's proto comment already documented.

## Changes

- `packages/proto/proto/adopt_dont_shop/applications/v1/application.proto` — `CountAdoptedAdoptersRequest`: renamed `created_after`/`created_before` → `actioned_after`/`actioned_before`, added optional `rescue_id`, documented the 10,000-entry `adopter_ids` cap. Regenerated `packages/proto/src/generated/**`. Also corrected the RPC doc comment's stated permission (was `applications.read`, actually `ANALYTICS_VIEW`/`admin.reports` — Copilot review catch).
- `services/applications/src/grpc/attribution-handlers.ts` — added the `MAX_ADOPTER_IDS` (10,000) cap check, `Date.parse` validation for both bounds, an `actioned_after > actioned_before` check, optional `rescue_id` scoping (`AND rescue_id = $N`), and switched the query to filter `actioned_at`. `MIN_COHORT_SIZE` k-anonymity floor from ADS-1006 is unchanged.
- `services/applications/src/grpc/event-store.ts` — `projectReadModel` now stamps `applications.actioned_at` from `state.decidedAt` (on `approved`) / `state.adoptedAt` (on `adopted`) — both already carry the triggering event's `occurred_at` — and leaves it `null` for every other status (rejected/withdrawn never qualify for attribution).
- `services/applications/src/migrations/010_backfill_application_actioned_at.ts` (new) — backfills `actioned_at` for existing approved/adopted rows from the `application_events` store; `up`/`down` both included. Casts `applications.status::text` when comparing to `application_events.event_type` (varchar) — Postgres rejects the bare enum = varchar comparison.
- `services/rescue/src/grpc/event-handlers.ts` — `countAdoptionsFromEvent` passes the event's own `rescue_id`, renames the date fields, and bounds the attendee fetch at `ATTRIBUTION_MAX_COHORT` (10,000, mirrors `MAX_ADOPTER_IDS` by value) — an event past that size fails the call with `INVALID_ARGUMENT` rather than silently truncating the registrant list.
- Tests updated/added in `attribution-handlers.test.ts`, `review-handlers.test.ts`, `event-handlers.test.ts`, `applications-client.test.ts`, `migrations.test.ts`.

**Not done, flagged instead of done silently:** the existing `applications_rescue_status_created_idx` index is on `(rescue_id, status, created_at)`, not `actioned_at`, so the new rescue-scoped attribution query doesn't get full index coverage from it. A matching `(rescue_id, status, actioned_at)` index looks like a reasonable follow-up but felt like scope creep for this change (attribution queries are low-volume admin/reporting calls, not a hot path) — left a code comment at the query site and flagging it here rather than adding an index quietly.

## Before requesting review

- [x] Ran the scoped verification for the touched packages: `pnpm exec turbo test type-check lint --filter=@adopt-dont-shop/service.applications --filter=@adopt-dont-shop/service.rescue --filter=@adopt-dont-shop/proto` (all green — see Test plan). Did not run the repo-wide `pnpm ci:local:quick`.
- [x] Tests for new behaviour added (TDD)
- [ ] If touching `.env` requirements, updated `.env.example`'s REQUIRED block — n/a, no new env vars
- [ ] If touching a `lib.*`, considered consumer impact across `app.*` and `service.backend` — n/a, no `lib.*` touched

## Test plan

- [x] Existing tests pass — `pnpm exec turbo test --filter=@adopt-dont-shop/service.applications --filter=@adopt-dont-shop/service.rescue --filter=@adopt-dont-shop/proto` → 308 + 256 + 66 tests passing
- [x] New behaviour is covered by tests — rescue scoping (inclusion/exclusion), cap rejection, invalid-date rejection, `actioned_after > actioned_before` rejection, the equal-bound boundary case, `actioned_at` stamped on approve/re-stamped on markAdopted/left null on reject, event-side attendee-cap rejection, `rescue_id` forwarded from `getEventAnalytics`
- [x] Tested manually — found a local Postgres 16 cluster in the sandbox, started it, and ran the actual `node-pg-migrate` runner (not just the SQL by eye) through a full up → down → up cycle against real test data (an approved row, an adopted row with both an `approved` and a later `adopted` event, and a `rejected` row): confirmed `actioned_at` is set to the correct event's `occurred_at`, `adopted` correctly wins over the earlier `approved` timestamp, `rejected` is never touched, the backfill is idempotent on re-run, and `down` clears exactly what `up` set
- [x] No TypeScript errors — `pnpm exec turbo type-check --filter=@adopt-dont-shop/service.applications --filter=@adopt-dont-shop/service.rescue --filter=@adopt-dont-shop/proto`
- [x] Lint passes — `pnpm exec turbo lint --filter=@adopt-dont-shop/service.applications --filter=@adopt-dont-shop/service.rescue --filter=@adopt-dont-shop/proto` (one pre-existing unrelated warning in `pets-client.test.ts`, not touched by this change)

## Related

- ADS-1023, ADS-1024, ADS-1025

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01PyLUww6MdfCdqbXe9V6eAE

### CODE REVIEW

File: packages/proto/proto/adopt_dont_shop/applications/v1/application.proto [hunk 1]
```diff
@@ -96,11 +96,15 @@ service ApplicationService {
...
-  // involved. Caller MUST hold applications.read.
+  // ADOPTED status with actioned_at (ADS-1025 — when the application
+  // reached that status, not when it was created) inside
+  // [actioned_after, actioned_before], optionally narrowed to one
+  // rescue (ADS-1023). Deliberately narrow: the caller already knows
+  // the adopter_ids (it resolved them itself); the response reveals
+  // only the count, never which ids matched or which rescue/pet was
+  // involved. Caller MUST hold applications.read. adopter_ids is
+  // capped at 10,000 entries (ADS-1024) — rejected with
+  // INVALID_ARGUMENT above that.
```
[the remark this answers is not in the document]
Comment (ParagonJenko) [reply packages/proto/proto/adopt_dont_shop/applications/v1/application.proto]: Good catch — corrected the comment to describe the real gate: `ANALYTICS_VIEW` (`admin.reports`), an admin-only reporting permission, not `applications.read`. Regenerated the proto codegen (`pnpm exec turbo build --filter=@adopt-dont-shop/proto`) so the fix propagates into the generated TS, and pushed both in `655ecb6`.

---
_Generated by [Claude Code](https://claude.ai/code)_
